frontend/src/main.js

```diff
@@ -104,6 +104,16 @@ Vue.prototype.$config = {
     if (result.data.ux) {
       Object.assign(Vue.prototype.$config.ux, result.data.ux);
     }
+  }
+
+  //Tails configured?
+  result = await axios
+    .get(`${apiBaseUrl}/admin/config/tailsConfigured`)
+    .catch((e) => {
+      console.error(e);
+    });
+  if ({}.hasOwnProperty.call(result, "data")) {
+    Vue.prototype.$config.tailsConfigured = result.data.tailsConfigured;
     console.log("...Configuration loaded");
   }
```

Comment:
Isn't the whole config already fetched in line 96 ? Is `tailsConfigured` not part of this object?

Reply:
no, but maybe we could make it part of that config.
the other config stuff is all static config data of the bpa. this here is queried from the acapy.
I could also merge this if @etschelp  tells me how the live cycle of this config Object in the backend is....

Reply:
@etschelp what I mean is RuntimeConfig is more about static properties - at least up to now....

Reply:
ah, I see. I'd either make it part of config or part of settings

Reply:
I would add this to the RuntimeConfig there you have multiple options 

1. Hack it like

```java

    Boolean tailsServerConfigured;

    @Inject
    public RuntimeConfig(AriesClient ac) {
        try {
            this.tailsServerConfigured = ac.statusConfig().flatMap(c -> c.getAs("tails_server_base_url", String.class)).isPresent();
        } catch (IOException e) {
            log.warn("No aca-py");
        }
    }
```

With the drawback that tailsServerConfigured will be null if aca-py is not immediately available after startup. You can test like this but it is not a good solution.

2. Better

Add an event emitter to the startup task that emits an aca-py ready event and do whats been done in the constructor above in the event handler.

https://docs.micronaut.io/latest/guide/#contextEvents

StartupTasks
```java
        ac.statusWaitUntilReady(Duration.ofSeconds(60));
        eventPublisher.publishEvent(new AcaPyReady());

        @NoArgsConstructor
        public static final class AcaPyReady{};
```

```java
public class RuntimeConfig implements ApplicationEventListener<StartupTasks.AcaPyReady> {

    @JsonIgnore
    @Inject
    AriesClient ac;
    
    Boolean tailsServerConfigured;

    @Override
    public void onApplicationEvent(StartupTasks.AcaPyReady event) {
        try {
            this.tailsServerConfigured = ac.statusConfig().flatMap(c -> c.getAs("tails_server_base_url", String.class)).isPresent();
        } catch (IOException e) {
            log.warn("No aca-py");
        }
    }
}
```
3. Create a service class that sets the missing config values if null.

Reply:
@etschelp thanks for the suggestions . I implemented it as you proposed.
Nevertheless I still ask myself if we shouldn't wait for the acapy synchronously and block the complete frontend until its loaded. Why?
- without the acapy we cannot do much in the UI
- it won't happen often that the user really has to wait for the acapy ( only during upgrades maybe...)
--> so do we really have to do this asynchronously?